src/ImportDefinitionsBundle/Export/Exporter.php

```diff
@@ -0,0 +1,266 @@
+        $fetcher = $this->getFetcher($definition);
+        $total = $fetcher->count($definition, $params);
+
+        if ($total > 0) {
+            $this->eventDispatcher->dispatch('export_definition.total', new ImportDefinitionEvent($definition, $total));
+
+            $this->runExport($definition, $params, $total, $fetcher);
+        }
+
+        $this->eventDispatcher->dispatch('export_definition.finished', new ImportDefinitionEvent($definition));
+    }
+
+    /**
+     * @param DefinitionInterface $definition
+     * @return FetcherInterface
+     */
+    private function getFetcher(DefinitionInterface $definition)
+    {
+        if (!$this->fetcherRegistry->has($definition->getFetcher())) {
+            throw new \InvalidArgumentException(sprintf('Definition %s has no valid fetcher configured', $definition->getName()));
+        }
+
+        /** @var FetcherInterface $fetcher */
+        $fetcher = $this->fetcherRegistry->get($definition->getFetcher());
+
+        return $fetcher;
+    }
+
+    /**
+     * @param DefinitionInterface $definition
+     * @param                     $params
+     * @param int                 $total
+     * @param FetcherInterface    $fetcher
+     * @throws \Exception
+     */
+    private function runExport(DefinitionInterface $definition, $params, int $total, FetcherInterface $fetcher)
+    {
+        $count = 0;
+        $countToClean = 1000;
+        $perLoop = 50;
+
+        $entries = [];
+
+        for ($i = 0; $i < (ceil($total / $perLoop)); $i++) {
+
+            $objects = $fetcher->fetch($definition, $params, $perLoop, $i * $perLoop);
+
+            foreach ($objects as $object) {
+                try {
+                    $entries[] = $this->exportRow($definition, $object, $params);
+
+                    if (($count + 1) % $countToClean === 0) {
+                        \Pimcore::collectGarbage();
+                        $this->logger->info('Clean Garbage');
+                        $this->eventDispatcher->dispatch(
+                            'export_definition.status',
+                            new ImportDefinitionEvent($definition, 'Collect Garbage')
+                        );
+                    }
+
+                    $count++;
+                } catch (\Exception $ex) {
+                    $this->logger->error($ex);
+
+                    $this->exceptions[] = $ex;
+
+                    $this->eventDispatcher->dispatch(
+                        'export_definition.status',
+                        new ImportDefinitionEvent($definition, sprintf('Error: %s', $ex->getMessage()))
+                    );
+
+                    if ($definition->getStopOnException()) {
+                        throw $ex;
+                    }
+                }
+
+                $this->eventDispatcher->dispatch(
+                    'export_definition.progress',
+                    new ImportDefinitionEvent($definition)
+                );
+            }
+        }
+
+        print_r($entries);
```

Comment:
Oops. :)

Reply:
WIP ;), persisting is not yet implemented.